packages/gcloud-mcp/src/windows_gcloud_utils.ts

```diff
@@ -0,0 +1,184 @@
+}
+
+
+export function runWhere(command: string, spawnEnv: {[key: string]: string|undefined}): string[] {
+    try {
+        const result = child_process
+                            .execSync(`where ${command}`, {
+                                encoding: 'utf8',
+                                stdio: ['pipe', 'pipe', 'ignore'],
+                                env: spawnEnv,  // Use updated PATH for where command
+                            })
+                            .trim();
+        return result.split(/\r?\n/).filter(line => line.length > 0);
+    } catch (e) {
+        return [];
+    }
+}
+
+export function getPythonVersion(pythonPath: string, spawnEnv: {[key: string]: string|undefined}): string | undefined {
+    try {
+        const escapedPath =
+            pythonPath.includes(' ') ? `"${pythonPath}"` : pythonPath;
+        const cmd = `${escapedPath} -c "import sys; print(sys.version)"`;
+        const result = child_process
+                            .execSync(cmd, {
+                                encoding: 'utf8',
+                                stdio: ['pipe', 'pipe', 'ignore'],
+                                env: spawnEnv,  // Use env without PYTHONHOME
+                            })
+                            .trim();
+        return result.split(/[\r\n]+/)[0];
+   } catch (e) {
+     return undefined;
+   }
+}
+
+export function findWindowsPythonPath(spawnEnv: {[key: string]: string|undefined}): string {
+    // Try to find a Python installation on Windows
+    // Try Python, python3, python2
+    
+    const pythonCandidates = runWhere('python', spawnEnv);
+    if (pythonCandidates.length > 0) {
+        for (const candidate of pythonCandidates) {
+            const version = getPythonVersion(candidate, spawnEnv);
+            if (version && version.startsWith('3')) {
+                return candidate;
+         }
+        }
+    }   
+
+    const python3Candidates = runWhere('python3', spawnEnv);
+    if (python3Candidates.length > 0) {
+        for (const candidate of python3Candidates) {
+            const version = getPythonVersion(candidate, spawnEnv);
+            if (version && version.startsWith('3')) {
+                return candidate;
+         }
+        }
+    }
+
+    // Try to find python2 last
+    if (pythonCandidates.length > 0) {
+        for (const candidate of pythonCandidates) {
+            const version = getPythonVersion(candidate, spawnEnv);
+            if (version && version.startsWith('2')) {
+                return candidate;
+            }
+        }
+    } 
+    return "python.exe"; // Fallback to default python command
+}
+
+
+export function getCloudSDKSettings(
+    currentEnv: NodeJS.ProcessEnv = process.env,
+    scriptDir: string = __dirname,
+): CloudSdkSettings {
+    const env = {...currentEnv };
+    let cloudSdkRootDir = env['CLOUDSDK_ROOT_DIR'] || '';
+    if (!cloudSdkRootDir) {
+        cloudSdkRootDir = path.resolve(scriptDir, '..');
```

Comment:
This logic was originally written from the perspective of the gcloud.cmd file. The `__dirname/..` default is not correct. 

Instead: `[where gcloud]/../bin/sdk`